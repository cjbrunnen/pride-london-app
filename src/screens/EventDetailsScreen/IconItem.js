// @flow
import React from "react";
import { View, StyleSheet } from "react-native";
import type { Node } from "react";
import Text from "../../components/Text";
import type { TextType } from "../../components/Text";

type IconItemProps = {|
  icon: Node,
  title: string,
  content?: Node,
  titleType?: TextType
|};

const IconItem = ({ icon, title, titleType, content }: IconItemProps) => (
  <View style={styles.iconItem}>
    <View style={styles.icon}>{icon}</View>
    <View style={styles.item}>
      <Text type={titleType}>{title}</Text>
      {content}
    </View>
  </View>
);
IconItem.defaultProps = {
  content: null,
  titleType: "h3"
};

const styles = StyleSheet.create({
  iconItem: {
    flex: 1,
    flexDirection: "row"
  },
  icon: {
    width: 40,
    height: 20,
    paddingRight: 20
  },
  item: {
    flex: 1
  }
});

export default IconItem;
