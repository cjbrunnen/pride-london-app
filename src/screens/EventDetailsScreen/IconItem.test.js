// @flow
import React from "react";
import { Text } from "react-native";
import { shallow } from "enzyme";
import IconItem from "./IconItem";

it("renders correctly", () => {
  const output = shallow(
    <IconItem icon={<Text>Some icon</Text>} title="some thing" />
  );
  expect(output).toMatchSnapshot();
});

it("renders content", () => {
  const output = shallow(
    <IconItem
      icon={<Text>Some icon</Text>}
      title="some thing"
      content={<Text>Some child</Text>}
    />
  );

  const texts = output.find("Text");

  const hasChild = texts.someWhere(
    element => element.children().text() === "Some child"
  );

  expect(hasChild).toBe(true);
});

it("passes titleType to title", () => {
  const output = shallow(
    <IconItem icon={<Text>Some icon</Text>} title="some thing" titleType="h1" />
  );

  const texts = output.find("Text");

  const hasType = texts.someWhere(element => element.props().type === "h1");

  expect(hasType).toBe(true);
});
