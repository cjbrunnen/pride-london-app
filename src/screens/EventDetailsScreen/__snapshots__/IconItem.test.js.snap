// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "height": 20,
        "paddingRight": 20,
        "width": 40,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Some icon
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      markdown={false}
      style={Object {}}
      type="h3"
    >
      some thing
    </Text>
  </View>
</View>
`;
